sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FilterPushdownSuite.scala

```diff
@@ -1016,6 +1016,8 @@ class FilterPushdownSuite extends PlanTest {
     val correctAnswer = x.where("x.a".attr === 5).join(y.where("y.a".attr === 5),
         condition = Some("x.a".attr === Rand(10) && "y.b".attr === 5))
 
-    comparePlans(Optimize.execute(originalQuery.analyze), correctAnswer.analyze)
```

Comment:
```SQL
select actno from act where rand() < 0.5 and rand() >= 0.5
```

Based on your conclusion, this should never return any value, right? However, I checked the other DBMS and the filter conditions are not always false.